native-maven-plugin/src/main/java/org/graalvm/buildtools/maven/sbom/SBOMGenerator.java

```diff
@@ -0,0 +1,283 @@
+         */
+        static final String packageNames = "packageNames";
+        /**
+         * The path to the jar containing the class files. For a component embedded in a shaded jar, the path must
+         * be pointing to the shaded jar.
+         */
+        static final String jarPath = "jarPath";
+        /**
+         * If set to false, then this component and all its transitive dependencies SHOULD NOT be pruned by Native Image.
+         * This is set to false when the package names could not be derived accurately.
+         */
+        static final String prunable = "prunable";
+    }
+
+    public SBOMGenerator(
+            MavenProject mavenProject,
+            MavenSession mavenSession,
+            BuildPluginManager pluginManager,
+            RepositorySystem repositorySystem,
+            String mainClass,
+            Logger logger) {
+        this.mavenProject = mavenProject;
+        this.mavenSession = mavenSession;
+        this.pluginManager = pluginManager;
+        this.repositorySystem = repositorySystem;
+        this.mainClass = mainClass;
+        this.logger = logger;
+    }
+
+    /**
+     * Checks if the JDK version supports augmented SBOMs.
+     *
+     * @param detectedJdkVersion the JDK version used.
+     * @param throwErrorIfNotSupported if true, then an error is thrown if the check failed.
+     * @return true if the JDK version supports the flag, otherwise false (if {@param throwErrorIfNotSupported} is false).
+     * @throws IllegalArgumentException when {@param throwErrorIfNotSupported} is true and the version check failed.
+     */
+    public static boolean checkAugmentedSBOMSupportedByJDKVersion(int detectedJdkVersion, boolean throwErrorIfNotSupported) throws IllegalArgumentException {
+        if (detectedJdkVersion < SBOMGenerator.requiredNativeImageVersion) {
+            if (throwErrorIfNotSupported) {
+                throw new IllegalArgumentException(
+                        String.format("%s version %s is required to use configuration option %s but major JDK version %s has been detected.",
+                                ORACLE_GRAALVM_IDENTIFIER, SBOMGenerator.requiredNativeImageVersion, augmentedSBOMParamName, detectedJdkVersion));
+            }
+            return false;
+        }
+        return true;
+    }
+
+    /**
+     * Generates an SBOM that will be further augmented by Native Image. The SBOM is stored in the build directory.
+     *
+     * @throws MojoExecutionException if SBOM creation fails.
+     */
+    public void generate() throws MojoExecutionException {
+        String outputDirectory = mavenProject.getBuild().getDirectory();
+        Path sbomPath = Paths.get(outputDirectory, SBOM_NAME + "." + FILE_FORMAT);
+        try {
+            /* Suppress the output from the cyclonedx-maven-plugin. */
+            int loggingLevel = logger.getThreshold();
+            logger.setThreshold(Logger.LEVEL_DISABLED);
+            executeMojo(
```

Comment:
:x: This should ideally be done without using an external plugin to invoke the mojo. We do this already in several plugins.

Reply:
Ah, great! Could you give me some pointers?

Reply:
Actually made a mistake, the code I had in mind (Micronaut Maven plugin) is using the same plugin :man_facepalming: So consider my comment resolved!